.github/workflows/lighthouse-ci.yml

```
@@ -0,0 +1,14 @@
+name: Lighthouse CI
+on: push
```

@minimalsm Do you know which pushes this refers to? We don't seem to be specifying a branch so I'd assume this is any push to any branch which we may not want. Should this be only on pushes to `master` since we're checking the production link below?

---

@wackerow I hadn't worked that part out yet 😁 any takes?

---

Agreed. I wonder if there is a way to trigger this in a more "on-demand" way (e.g., in branches where we are doing performance improvements would be nice to have these metrics).

---

Looking through this [Github Actions Documentation](https://github.com/ethereum/ethereum-org-website/pull/12015)

Maybe look for branches with a specific prefix, like `performance`, during a pull request or push to an intermediary maybe?

```suggestion
on: 
  pull_request:
    branches:
      - 'performance/**'
```

So if we want to check a performance update, create the branch with this specific prefix to push to a PR.

Haven't seen anything that would make this more dependent on some other automated process. (Rather, not rely on having to remember to create a specific branch name, say)

---

That is an easy to implement idea! good for testing purposes.

---

Yeah, that's a great idea... perhaps also `master`? I think a standard check of every page on push to master would be great, but then also the ability to trigger a branch by using `performance/**` would give us the flexibility and consistency we need.

In the meantime, Netlify is running Lighthouse on every build, but only for the Homepage... is that correct @minimalsm?

---

Correct. The Netlify plugin runs just for the home page.

Ok, yeah, I think we have agreed on adding `master`, `staging`, and `performance/**` for now. Will do those changes and get this merged.

---

@pettinarip throwing this out there for at least future consideration.

What about triggering based on a label added? So have a `Performance` label?

This will probably need to be manually added to the PR.

---

Thats another great idea! however, we are planning on running this on all the PRs once we finish testing them. Depending on what we decided at the end, I think this could be a nice idea to keep in mind, just in case.